exercicios/para-casa/LohanaR/atividadepracasa.py

```diff
@@ -0,0 +1,132 @@
+def main():
+    dataset = {'discentes': []}
+    print("\n---  Olá, seja bem-vinde à Escola do Reprograma!  ---")
+    print("Sistema de informações de discentes")
+
+    while True:
+        cod_opcao = obter_opcao()
+        if cod_opcao == 1:
+            incluir_nova_discente(dataset)
+        elif cod_opcao == 2:
+            consultar_lista_discentes(dataset)
+        elif cod_opcao == 3:
+            consultar_faltas_discente(dataset)
+        elif cod_opcao == 4:
+            consultar_notas_discente(dataset)
+        elif cod_opcao == 5:
+            consultar_status_aprovacao(dataset)
+        elif cod_opcao == 6:
+            print("Encerrando o programa...")
+            encerramento()
+            break
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                     "1 - Incluir uma nova aluna\n"
+                                     "2 - Consultar lista de alunas\n"
+                                     "3 - Consultar faltas da aluna\n"
+                                     "4 - Consultar notas da aluna\n"
+                                     "5 - Consultar status de aprovação\n"
+                                     "6 - Sair do sistema\n"
+                                     "Opção: "))
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+    return codigo_opcao
+
+def incluir_nova_discente(dataset):
+    nome = input("Por favor, digite o nome da discente: ")
+    try:
+        faltas = int(input("Por favor, digite o número de faltas: "))
+    except ValueError:
+        print("Esse número de faltas é inválido. Por favor, digite um número inteiro.")
+        return
+
+    notas = []
+    print("Por favor, digite as notas da discente. Se desejar encerrar, digite 'finalizar' para concluir.")
+    while True:
+        nota = input("Nota: ")
+        if nota.lower() == 'finalizar':
+            break
+        try:
+            nota_float = float(nota)
+            if 0 <= nota_float <= 10: 
+                notas.append(nota_float)
+            else:
+                print("Essa nota não atende o critério do intervalo aceitável (0 a 10).")
+        except ValueError:
+            print("Essa entrada é inválida. Por favor, digite um número válido ou 'finalizar' para concluir.")
+
+    if len(notas) == 0:
+        print("Discente não incluída.")
+        return
+
+    dataset['discentes'].append({
+        'nome': nome,
+        'faltas': faltas,
+        'notas': notas
+    })
```

Comment:
Dentro da proposta do exercício era utilizar a tupla como estrutura de chave para o dicionário. Não foi utilizado nem o dataset proposto, e o dataset montado não segue a estrutura proposta. 

**Sugestão:**
```suggestion
def salvar_dados(nome, sobrenome, turma, lista_presença, nota_participação, notas):
    chave = (nome, sobrenome)
    dataset[chave] = { 
        "Turma": turma,
        "Notas": notas,
        "Presença": lista_presença,
        "Participação": nota_participação
    }
```